src/lowlevel/connection.rs

```diff
@@ -63,24 +67,24 @@ impl<W, Buffer, Auxiliary> Drop for SharedData<W, Buffer, Auxiliary> {
 }
 
 impl<W, Buffer, Auxiliary> SharedData<W, Buffer, Auxiliary> {
-    pub(crate) fn writer(&self) -> &WriterBuffered<W> {
-        &self.0.writer
+    pub(crate) fn writer(&self) -> Pin<&WriterBuffered<W>> {
```

Comment:
This is a weird type. In general, pinning a shared type is of little value, since `&` is inherently already not going to move.

Reply:
The problem here is `WriteBuffered` stored a `PinnedMutexAsync<impl AsyncWrite>`.

If we do not return a `Pin`ed type here, then I have to use `unsafe` [`Pin::new_unchecked`](https://doc.rust-lang.org/std/pin/struct.Pin.html#method.new_unchecked) to manually create a `Pin`ned mutable reference to the `impl AsyncWrite` object.

Reply:
See my comment in `pin_util`.